connPFM/connectivity/ev.py

```diff
@@ -68,132 +40,128 @@ def event_detection(data_file, atlas, surrprefix="", sursufix="", nsur=100, segm
-
-    p = np.zeros([t, 1])
-    rssr_flat = rssr.flatten()
-    for i in range(t):
-        p[i] = np.mean(rssr_flat >= rss[i])
-    # apply statistical cutoff
-    pcrit = 0.001
-
-    # find frames that pass statistical testz_ts
-    idx = np.argwhere(p < pcrit)[:, 0]
-    if segments:
-        # identify contiguous segments of frames that pass statistical test
-        dff = idx.T - range(len(idx))
-        unq = np.unique(dff)
-        nevents = len(unq)
-
-        # find the peak rss within each segment
-        idxpeak = np.zeros([nevents, 1])
-        for ievent in range(nevents):
-            idxevent = idx[dff.T == unq[ievent].T]
-            rssevent = rss[idxevent]
-            idxmax = np.argmax(rssevent)
-            idxpeak[ievent] = idxevent[idxmax]
-        idxpeak = idxpeak[:, 0].astype(int)
-    # get activity at peak
-    else:
-        idxpeak = idx
-    tspeaks = z_ts[idxpeak, :]
+            rssr[:, irand] = surrogate_events[irand][0]
+
+        # TODO: find out why there is such a big peak on time-point 0 for AUC surrogates
+        if "AUC" in surrprefix:
+            rssr[0, :] = 0
+
+        # Initialize p-values array with ones to avoid default values being lower
+        # than the threshold p-value
+        p_value = np.ones(t)
+
+        # Statistical cutoff
+        pcrit = 0.001
+
+        # Find peaks with a general threshold for RSS
+        if peak_detection == "rss":
+            rssr_flat = rssr.flatten()
+            for i in range(t):
+                p_value[i] = np.mean(rssr_flat >= rss[i])
+
+        # Find peaks with a different threshold for each time-point in RSS
+        elif peak_detection == "rss_time":
+            for i in range(t):
+                p_value[i] = np.mean(np.squeeze(rssr[i, :]) >= rss[i])
+
+        # find frames that pass statistical testz_ts
+        idx = np.argwhere(p_value < pcrit)[:, 0]
+        if segments:
+            idxpeak = connectivity_utils.remove_neighboring_peaks(rss, idx)
+        # get activity at peak
+        else:
+            idxpeak = idx
+
+        # get co-fluctuation at peaks
+        etspeaks = connectivity_utils.threshold_ets_matrix(ets, thr=0, selected_idxs=idxpeak)
+
+    # Make selection of points with edge time-series matrix
+    elif "ets" in peak_detection:
+        LGR.info("Selecting points with edge time-series matrix...")
+        if peak_detection == "ets":
+            LGR.info("Reading AUC of surrogates to perform the thresholding step...")
+            thr = connectivity_utils.surrogates_histogram(
+                surrprefix,
+                sursufix,
+                masker,
+                hist_range=(hist_min, hist_max),
+                numrand=nsur,
+            )
+        elif peak_detection == "ets_time":
+            # Initialize array for threshold
+            thr = np.zeros(t)
+
+            # Calculate ETS matrix of each surrogate
+            sur_ets = Parallel(n_jobs=-1, backend="multiprocessing")(
```

Comment:
suggestion, In the function `rss_surr` we already calculate ets of each surrogate to get the rss, can't you take it from that function. That way we are computationally more efficient and we avoid duplicating code

Reply:
Good point! Didn't see that. I will update `rss_surr` to return the ets matrix.

Reply:
`875c96b` should do this.